graylog2-web-interface/src/components/metrics/CounterRate.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import numeral from 'numeral';
+
+var CounterRate = React.createClass({
+  propTypes: {
+    metric: React.PropTypes.object,
+    showTotal: React.PropTypes.bool,
+    prefix: React.PropTypes.string,
+    suffix: React.PropTypes.string,
+    hideOnZero: React.PropTypes.bool,
+    hideOnMissing: React.PropTypes.bool,
+  },
+  getDefaultProps() {
+    return {
+      showTotal: false,
+      prefix: null,
+      suffix: 'per second',
+      hideOnZero: false,
+      hideOnMissing: false,
+    };
+  },
+
+  getInitialState() {
+    return {
+      prevMetric: null,
+      prevTs: null,
+      nowTs: this._nowInSeconds(),
+    };
+  },
+  componentWillReceiveProps() {
+    this.setState({
+      prevMetric: this.props.metric,
+      prevTs: this.state.nowTs,
+      nowTs: this._nowInSeconds(),
+    });
+  },
+
+  _nowInSeconds() {
+    return Math.floor(Date.now() / 1000);
+  },
+
+  _checkPrevMetric() {
+    return this.state.prevMetric && this.state.prevMetric.count !== undefined && this.state.prevTs;
+  },
+
+  _placeholder() {
+    if (this.props.hideOnZero) {
+      return null;
+    }
+    return (<span>{this._prefix()}Calculating...</span>);
+  },
+
+  _prefix() {
+    if (this.props.prefix) {
+      return this.props.prefix + ' ';
```

Comment:
Please use templates for concatenating strings: http://eslint.org/docs/rules/prefer-template